packages/shared/src/util/features.ts

```diff
@@ -726,3 +745,97 @@ export function getParsedPrereqCondition(condition: string) {
   }
   return undefined;
 }
+function checkEnvironmentsMatch(
+  feature: FeatureInterface,
+  environments: string[],
+  reviewSetting: RequireReview
+) {
+  for (const env of reviewSetting.environments) {
+    if (environments.includes(env)) {
+      return true;
+    }
+  }
+  // check projects
+  if (feature?.project && reviewSetting.projects.includes(feature?.project)) {
+    return true;
+  }
+  for (const tag of reviewSetting.tags) {
+    if (feature?.tags?.includes(tag)) {
+      return true;
+    }
+  }
+  return (
+    reviewSetting.environments.length === 0 &&
+    reviewSetting.projects.length === 0 &&
+    reviewSetting.tags.length === 0
+  ); //if everything is empty is means it is on for all.
+}
+export function featureRequiresReview(
+  feature: FeatureInterface,
+  requestedEnvironments: string[],
+  settings?: OrganizationSettings
+) {
+  const requiresReviewSettings = settings?.requireReviews;
+  //legacy check
+  if (
+    requiresReviewSettings === undefined ||
+    requiresReviewSettings === true ||
+    requiresReviewSettings === false
+  ) {
+    return !!requiresReviewSettings;
+  }
+
+  for (const reviewSetting of requiresReviewSettings) {
+    if (reviewSetting.requireReviewOn === true) {
+      return checkEnvironmentsMatch(
+        feature,
+        requestedEnvironments,
+        reviewSetting
+      );
+    }
+  }
+
+  return false;
+}
+
+export function resetReviewOnChange(
+  feature: FeatureInterface,
+  requestedEnvironments: string[],
+  settings?: OrganizationSettings
+) {
+  const requiresReviewSettings = settings?.requireReviews;
+  //legacy check
+  if (
+    requiresReviewSettings === true ||
+    requiresReviewSettings === false ||
+    requiresReviewSettings === undefined
+  ) {
+    return !!requiresReviewSettings;
+  }
+  for (const reviewSetting of requiresReviewSettings) {
+    if (reviewSetting.resetReviewOnChange === true) {
+      return checkEnvironmentsMatch(
+        feature,
+        requestedEnvironments,
+        reviewSetting
+      );
+    }
```

Comment:
This logic doesn't seem right.  I think we'd want to find the first review setting that matches the feature and return the `resetReviewOnChange` setting from that.